File: backend/app/schemas/drive.py
from pydantic import BaseModel


class DriveFolder(BaseModel):
    id: str
    name: str


class DriveFileListing(BaseModel):
    id: str
    name: str
    size_bytes: int
    status_reason: str | None  # None | "multi_parent" | "no_parent"


class FolderConfig(BaseModel):
    folder_id: str
    folder_name: str
    created_by_app: bool


class SelectFolderRequest(BaseModel):
    folder_id: str


class CreateFolderRequest(BaseModel):
    name: str
    parent_id: str | None = None

File: frontend/src/types/drive.ts
export interface DriveFolder {
  id: string
  name: string
}

export type FileStatusReason = 'multi_parent' | 'no_parent' | null

export interface DriveFileListing {
  id: string
  name: string
  size_bytes: number
  status_reason: FileStatusReason
}

export interface FolderConfig {
  folder_id: string
  folder_name: string
  created_by_app: boolean
}
